pg_shard.c

```diff
@@ -378,6 +378,25 @@ ErrorIfQueryNotSupported(Query *queryTree)
 						errmsg("unsupported utility statement")));
 	}
 
+	/*
+	 * Reject subqueries which are in SELECT or WHERE clause.
+	 * Queries which include subqueries in FROM clauses are rejected below.
+	 */
+	if (queryTree->hasSubLinks == true)
+	{
+		ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+						errmsg("cannot perform distributed planning on this query"),
+						errdetail("Subqueries are currently unsupported.")));
+	}
+
+	/* reject queries which include CommonTableExpr */
+	if (queryTree->cteList != NIL)
+	{
+		ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+						errmsg("cannot perform distributed planning on this query"),
+						errdetail("Common table entries are currently unsupported.")));
```

Comment:
It's _Common table **expressions**_. So _Common table expressions are not supported in distributed queries._.